documentation/projects/proposals/document_all_media_properties/20240313-implementation_plan_document_media_properties_api.md

```diff
@@ -0,0 +1,210 @@
+| -------------- | ---- | ---------------------------------------------- |
+| `allows_blank` | bool | whether the field accepts blank strings        |
+| `allows_null`  | bool | whether the field accepts null values          |
+| `is_pk`        | bool | whether the field has a primary key constraint |
+| `needs_unique` | bool | whether the field has a unique index           |
+| `default`      | bool | the value to use if one is not provided        |
+| `help_text`    | str  | a human-readable description of the field      |
+
+[np]:
+  https://docs.djangoproject.com/en/4.0/ref/models/fields/#django.db.models.Field.default
+
+```{note}
+- The `is_relation` data-point is used to render two separate tables for
+  relation and value fields.
+- The `allows_null` data-point is not relevant for relations as they are always
+  nullable.
+- The `max_length` data-point is not captured because it is automatically
+  covered under the `db_type` data-point.
+- `is_pk`, `needs_unique`, `allows_null` and `allows_blank` are presented
+  collectively as "constraints" although not technically DB-level constraints.
+```
+
+The information extracted from this introspection will be stored in memory for
+the next stage.
+
+### Merge with additional notes
+
+Additional notes or hardcoded information may be extra information about a field
+such as known inconsistencies or special cases. This information, if documented
+using a consistent format, can be used to augment the information extracted in
+the previous step.
+
+We will document this information in the docstring of the model classes from
+where it can be accessed using `inspect.getdoc`. To ensure that these notes are
+co-located with the fields' code, we will parse docstrings of all ancestors and
+mixins of a model class.
+
+Additional information about fields will be specific inside the model docstring
+as valid YAML under the "Properties" heading. Note the format should be exactly
+as follows with "Properties" being the final subheading.
+
+```python
+class Media(AbstractMedia):
+    """
+    This is a generic media model. In the real world, assume this to be
+    ``Image`` or ``Audio``. It can have its own docstring here.
+
+    Properties
+    ==========
+    title: >-
+        the title of the media; This documentation can span multiple lines and
+        _can contain Markdown_ as it will be injected as-is into the doc page.
+    """
+
+    pass
+```
+
+### Generate documentation page
+
+With all information in-memory about the media properties, the output will be
+written to a page by the management command. This requires some restructuring in
+the pages created for the catalog media properties documentation.
+
+Similar to the catalog documentation, the output document will include the
+`preamble.md` file from `api/api/docs/media_properties/preamble.md`. This will
+be followed by the Markdown generated from the media properties as a table.
```

Comment:
Could you include the intended destination of this file? Like in the [catalog documentation](https://docs.openverse.org/_preview/3965/projects/proposals/document_all_media_properties/20230426-implementation_plan_documenting_media_properties.html#media-properties-md).

Reply:
Yes! Added in 416e492.